### PR TITLE
fix(auth): avoid redirect loop for customer domains

### PR DESCRIPTION
**Pull request by cathteng**
`is_auth_required` checks for `organization_slug` in `kwargs` to redirect to the organization login page if the user isn't logged in. This is easily parsed when we don't have customer domains but we need to manually add it to kwargs when customer domains is on. The parsing for this was done below the `is_auth_required` function, so we just need to move it above.

Before

https://github.com/user-attachments/assets/cd19f497-6c13-4cc4-acc3-925f2385e9e4

After

https://github.com/user-attachments/assets/af049e3b-e5cd-4aa9-9f62-353eb63b2d00

### CODE REVIEW

File: src/sentry/web/frontend/base.py [hunk 1]
```diff
@@ -378,6 +372,12 @@ def dispatch(self, request: HttpRequest, *args: Any, **kwargs: Any) -> HttpRespo
             # In customer domain contexts, we will need to pre-populate the organization_slug keyword argument.
             kwargs["organization_slug"] = organization_slug
 
+        if self.is_auth_required(request, *args, **kwargs):
```
Comment (saponifi3d) [inline src/sentry/web/frontend/base.py:375]: can you add a unit test that recreates the behavior? (if we've run into the bug once, we'll probably run into it again)

File: tests/sentry/web/frontend/test_auth_login.py [hunk 2]
```diff
@@ -176,6 +176,20 @@ def test_login_valid_credentials_with_org_and_customer_domains(self):
...
+        self.create_project(organization=org, name="project")
+        # load it once for test cookie
+        self.client.get(self.path)
+
+        project_path = reverse("project-details", kwargs={"project_slug": "project"})
+        resp = self.client.get(project_path, HTTP_HOST=f"{org.slug}.testserver")
+
+        assert resp.status_code == 302
+        # redirect to auth org login page by parsing customer domain
+        assert getattr(resp, "url", None) == reverse("sentry-auth-organization", args=[org.slug])
```
Comment (saponifi3d) [inline tests/sentry/web/frontend/test_auth_login.py:191]: can we also load the url here and ensure that we get a 200 request after redirecting? (want to ensure we don't end up in a redirect loop)

**Review by saponifi3d — APPROVED** [review]
fix lgtm - updating the test to ensure there isn't a redirect loop would be great!

File: tests/sentry/web/frontend/test_auth_login.py (hunk 2 above)
Comment (cathteng) [reply tests/sentry/web/frontend/test_auth_login.py]: done, i actually checked and the logic before the fix would lead to a 302 :O

**Pull request merged**